pkg/vtpm/swtpm-vtpm/src/main.go

```diff
@@ -75,19 +82,27 @@ func makeDirs(dir string) error {
 	return nil
 }
 
+func readPidFile(pidPath string) (int, error) {
+	pidStr, err := os.ReadFile(pidPath)
+	if err == nil {
+		pid, err := strconv.Atoi(strings.TrimSpace(string(pidStr)))
```

Comment:
Note that this err is local to this block, hence it will not be reported in the fmt.Errorf at the end.
You can declare pid, and then avoid the implicit re-declation of err.